### DISCUSSION TITLE
How to use CurlOnlineService

### DISCUSSION THREAD

**Discussion by cweiske** [gerbera/gerbera#3185] [Q&A] [answered]
In the source code I saw https://github.com/gerbera/gerbera/blob/master/src/content/onlineservice/curl_online_service.cc which is not mentioned in the documentation.

Is this something I can use to plug external services with a defined API into gerbera?
How can I use it?
Which API is expected from the online service?

**Comment by KarlStraussberger** [comment, marked as answer]
Yes, this is the last fragment of the online service support (Mediatomb supported 4 which are out of order now).

You can do everything by deriving from that class and you have to hook it into content_manager to trigger it.

**Reply by cweiske** [reply]
Ah, so this is not a generic online service implementation I can use directly via the gerbera configuration xml file, but an abstract base class that can be used by other C code classes?

**Reply by KarlStraussberger** [reply]
We may implement that generic parser some day. But without real requirement this is quite complicated.
